### ISSUE TITLE
Videos lost when scraping from local emulation station

### ISSUE THREAD

**Issue by penx** [muldjord/skyscraper#285]
**Describe the bug**

Videos are lost when scraping from local emulation station

**To Reproduce**

Using retropie, if I:

- quit emulationstation
- delete all gamelist.xml files from my roms folder
  - e.g. `cd RetroPie/roms && find . -type f -name 'gamelist.xml' -delete`
- insert a USB stick containing my ROMs, gamelist.xml and media folders
- note that all the gamelist.xml have been copied across from USB stick in to the `/home/pi/RetroPie/roms/*` folders, along with media/videos
- relaunch emulationstation and confirm videos are appearing
- quit emulationstation and launch retropie setup
- launch skyscraper config and scrape from EmulationStation Gamelist (Local) with Download videos enabled
- Generate gamelist
- exit config and relaunch emulationstation

..then I would expect to see my skyscraper metadata merged with the videos from my USB stick, but the videos have disappeared.

**Technical information**
 - Skyscraper version: 3.6.6
 - Internet connection: wired broadband
 - OS: RetroPie
 - OS Version: 4.7.1, fully updated

**Comment by penx** [comment]
perhaps the RetroPie setup script is set to scrape from `~/.emulationstation/gamelists/[SYSTEM_NAME]/gamelist.xml` and not from `/home/pi/RetroPie/roms/[SYSTEM_NAME]/gamelist.xml`?

**Comment by muldjord** [comment]
That might be it, you can select between those in the script. But I don't use the script myself (and I'm not the author of the script, so I can't remember exactly how it works in regards to the input folder), so can't test it. But I can test whether the video importing from an ES gamelist works. So I'll try that and get back to you.

**Comment by muldjord** [comment]
Ok, so I just tested it, and the ES video importing works as expected. Here's what I did:

- Scraped a platform with videos enabled
- Generated the gamelist for said platform with videos enabled
- Verified that the videos were exported to the `/home/pi/RetroPie/rom/PLATFORM/media/videos` folder
- Purged all videos for the platform from the Skyscraper resource cache with `Skyscraper -p PLATFORM --cache purge:video
- Verified that the video cache for the platform was now empty with `Skyscraper -p PLATFORM --cache show`
- Scraped the platform with the `esgamelist` module with videos enabled and verified `YES` for videos for all games
- Regenerated the gamelist for ES and verified `YES` for all games

So it must be something either with the script or some other configuration that makes Skyscraper miss them from your system. Skyscraper will look for the gamelist.xml in the folder configured with `-g PATH`. When the RetroPie Skyscraper script exits, it will show you what command-line it used. Look for that parameter. This is where your gamelist.xml needs to be, and the paths inside it must point to valid video files, of course.

**Comment by penx** [comment]
it's running:

`/opt/retropie/supplementary/skyscraper/Skyscraper -p amiga -g /home/pi/RetroPie/roms/amiga -o /home/pi/RetroPie/roms/amiga/media -s esgamelist --flags unattend,skipped,videos,nobrackets,relative`

I can see in `/home/pi/RetroPie/roms/amiga/gamelist.xml` that there are entries with videos, e.g. `<video>./media/videos/somegame.mp4</video>` where `somegame` matches the ROM file name and the mp4 exists on disk, but this video is not present when I recreate my games list

**Comment by penx** [comment]
I've just noticed a file at `~/.skyscraper/skipped-amiga-import.txt` that contains entries of these files, with "No returned matches", which I think was generated last night when I tried to run against screenscraper.fr, but they're down at the moment.

**Comment by penx** [comment]
my `~/.skyscraper/cache/amiga/videos/esgamelist` is empty

**Comment by muldjord** [comment]
> I've just noticed a file at `~/.skyscraper/skipped-amiga-import.txt` that contains entries of these files, with "No returned matches", which I think was generated last night when I tried to run against screenscraper.fr, but they're down at the moment.

This is probably because you tried to scrape it with the `-s import` module at some point. It's not related to this.

I think I know what's wrong. You probably forgot to enable the cache refresh option. I just tried with your exact command-line, and it loads everything from cache since refresh is not enabled. If I enable refresh (add the `--cache refresh` option) it will find the files as expected. That's my best guess.

So you need to do this:
- Enable cache refresh in the script
- Gather resources from the `esgamelist` module
- Regenerate the gamelist

**Comment by penx** [comment]
awesome, adding `--cache refresh` worked ❤️

**Comment by penx** [comment]
via the RetroPie setup script, it seems "Force cache refresh" defaults to Disabled, which doesn't seem to make much sense for the esgamelist scraper. Do you think this should be raised against the retropie setup script?

**Comment by muldjord** [comment]
I'm not really sure - refresh is set to disabled by default to save bandwidth for the online sources (an important default, especially for users who add more roms and regather resources often). It would require an entirely different set of settings for online vs offline sources and I believe that would confuse users even more, and be a bit of a mess UI-wise.

**Comment by penx** [comment]
I'd perhaps consider having skyscraper ignore the `--cache` setting when it uses esgamelist then (and default to refresh), but that's up to you!

Feel free to close this issue if you disagree 😄

**Comment by muldjord** [comment]
That would make sense. :) I'll consider it. Thanks for the suggestion.

**Comment by muldjord** [comment]
I've settled on the following solution.

- I've added a `cacheRefresh="BOOL"` option to `config.ini`
- This option has been set to `true` for the `esgamelist` and `import` modules in the default `config.ini` that is copied from the `config.ini.example` file when Skyscraper is first installed

This means that any new Skyscraper installation will have these options set for those modules by default. I think this is a good solution as I then don't force anything on the user as hardcoded options.
